mysql/errname.go

```diff
@@ -77,7 +77,7 @@ var MySQLErrName = map[uint16]string{
 	ErrTooLongIdent:                             "Identifier name '%-.100s' is too long",
 	ErrDupFieldName:                             "Duplicate column name '%-.192s'",
 	ErrDupKeyName:                               "Duplicate key name '%-.192s'",
-	ErrDupEntry:                                 "Duplicate entry '%-.192s' for key %d",
+	ErrDupEntry:                                 "Duplicate entry '%s' for key '%s'",
```

Comment:
Note that MySQL 8 would truncate the "entry" to 64 characters. I'm not sure if TiDB truncates the output using the `%-.192s` here or manually.

```sql
mysql> create table x(a varchar(512));
Query OK, 0 rows affected (0.03 sec)

mysql> insert into x values (repeat('a', 512)), (repeat('a', 512));
Query OK, 2 rows affected (0.01 sec)
Records: 2  Duplicates: 0  Warnings: 0

mysql> create unique index y on x(a);
ERROR 1062 (23000): Duplicate entry 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa' for key 'y'
```